.changelog/10423.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+proxycfg: reference to entry in map should not panic.
```

Comment:
I think since this is user-facing it should reference the behavior that's being fixed, how about:
```suggestion
proxycfg: avoid panic when transparent proxy upstream is added and then removed.
```